scripts/ci/prepare_tool_scripts.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+set -euo pipefail
+
+function prepare_tool_script() {
+    IMAGE="${1}"
+    VOLUME="${2}"
```

Comment:
I have a problem with this. I want to add terraform but it needs more volumes to be added i.e. auth script folder for every provider you want to use with terraform.

Reply:
So this `prepare_tool_script "hashicorp/terraform:latest" ".aws" ".azure" ".config/gcloud" terraform` would not work.

So `VOLUMES` varies and also `COMMAND`.

Reply:
The docker run command which I use for terraform is: `docker run --rm -it -v \${HOST_HOME}/.aws:/root/.aws -v \${HOST_HOME}/.azure:/root/.azure -v \${HOST_HOME}/.config:/root/.config -v \${HOST_AIRFLOW_SOURCES}:/opt/airflow -w /opt/airflow --env-file <(env | grep TF) ${TERRAFORM_IMAGE} \"\$@\"`

Reply:
@potiuk WDYT of having `function prepare_terraform_script` ?

Reply:
I think it woudl be easy enough (as a follow-up change :)) to turn VOLUME into list of volumes. And adding "env" or -w is also quite OK. I'a m happy to make it as a follow-up pr afterwards.

Reply:
Sure 'prepare_terraform_script" sounds even better :)

Reply:
Okay, I am gonna add it when I am adding terraform. :)